### PR TITLE
[CMD PARSING]

### LINKED ISSUE #70: [CMD PARSING]
Regler le caught exception : basic string (NICK sans rien, USEr m m m m)

### LINKED ISSUE #75: [NC]
image

Pb de buffer mal clean

### LINKED ISSUE #76: [LOOP] Pb de buffer
ok, après fix de la function ParseCommand, test des commandes seules qui demandent un arg, sans arg :

Marche : NICK USER PASS / TOPIC JOIN KILL KICK LIST MODE PART

Marche pas : INVITE (segfault), NOTICE (basic string erase), PRIVMSG (Caught exception : basic_string::substr: __pos (which is 1) > this->size() (which is 0) **interessant cest pas un pb de npo**s la)

**Nouvelle issue** : server qui renvoie que les commandes (hors registration) qui marchent sont des unknown command, alors que dans parse command, le print est identique (à voir si y a pas un \r ou \n qui se cache)

_Originally posted by @marineks in https://github.com/marineks/Ft_irc/issues/70#issuecomment-1407634024_

### PR DESCRIPTION
**Pull request by marineks**
Resolves #75, #70 

### ISSUE
A lot of segfault, caught exceptions when typing a command without any argument in NC. We did not see this error until now because we were testing w/ irssi, and irssi itself returns an error before the message is sent to our server so...

### SOLUTION
I improved the error handling in the function parseCommand (in parsing.cpp)
- Added case when there was a prefix
- Added case when there were no arguments

### WARNING
**Please test this in irssi before approving the PR:**

- [x]  Test a registration with a wrong password and not correct number of args for user 
- [x] For the following commands, test them without any argument after (even these are commands that should take args) :

- [x]  Registration : PASS, NICK, USER 
- [x]  OPER, KICK, INVITE, MODE, KILL
- [x]  JOIN, TOPIC, LIST, NAMES
- [x] NOTICE, PRIVMSG

Please not that NOTICE and PRIVMSG should bring a basic string exception that I did not correct because in your latest PR, the code has changed a lot so it seemed futile;

Also note that now, the registration should work in _any_ order (NICK, PASS, USER for instance), and that if you make a mistake in the process, should you correct it after, the welcome message should be sent ! (cf. PR on the register branch that was merged into main on Friday)
For instance:
```shell
PASS test
PASS good_pwd
NICK
NICK marine
USER m m
USER m m m m
```

### NEXT ? 

Keep in mind that this issue raised another one : I discovered that the buffer system we had was not proof-perfect! Still I want to do this PR because this new problem has nothing to do with the parsing.
Long story short : some buffers are not correctly concatenated or cleared (Idk yet), and in our switch, we have empty strings sent; as such we have a lot of spam of unknown commands on the client side!

### CODE REVIEW

**Comment by marineks** [comment]
**UPDATE :** I think I fixed the problem with the loop and the weird buffers. Here is a screenshot of nc working with simple commands (aka correct user behaviour)
Resolves #76 
image

**Comment by marineks** [comment]
Ici, screenshot d'un **registration process avec beaucoup d'erreurs données par le user**
image

**Comment by marineks** [comment]
Ici cas post-registration, où l'on voit que l'on a plus de problème de buffer ou de segfault sur des commandes sans message ou prefixe:
image
